app/components/channel_list/categories/categories.tsx

```diff
@@ -24,15 +25,17 @@ const styles = StyleSheet.create({
 
 const extractKey = (item: CategoryModel) => item.id;
 
-const Categories = ({categories, currentChannelId}: Props) => {
+const Categories = ({categories, currentChannelId, currentUserId}: Props) => {
     const intl = useIntl();
+
     const renderCategory = useCallback((data: {item: CategoryModel}) => {
         return (
             <>
                 <CategoryHeader category={data.item}/>
                 <CategoryBody
                     category={data.item}
                     currentChannelId={currentChannelId}
+                    currentUserId={currentUserId}
```

Comment:
If category body has already an index file that has to go either way to the database, why not get it from the database there, instead of drilling it down here?

Reply:
The idea was to prevent having to perform the same query (regardless of being cached) for each category when the parent can perform the query just once